kanso.json

```diff
@@ -433,12 +433,12 @@
                                 "rows": 3
                             },
                             "title": "Message",
-                            "description": "Text that will be sent out in the alert. You can refer to alert_name, num_reports_threshold, time_window_in_days, num_counted_reports (total number of reports within the time window) and new_reports (the counted reports which haven't been messaged about yet). Refer to the latest report as new_reports.0 (not new_reports[0] as in usual javascript! This is specific to the templates we use.), the previous report as new_reports.1, etc. E.g. \"{{num_counted_reports}} patients with {{alert_name}} in {{time_window_in_days}} days reported at {{new_reports.0.contact.parent.name}}. . New reports from: {{new_reports.0.contact.name}}, {{new_reports.1.contact.name}}, {{new_reports.2.contact.name}}.\""
+                            "description": "Text that will be sent out in the alert. You can refer to alert_name, num_reports_threshold, time_window_in_days, num_counted_reports (total number of reports within the time window) and new_reports (the counted reports which haven't been messaged about yet). Refer to the latest report as new_reports.0 (not new_reports[0] as in usual javascript! This is specific to the templates we use.), the previous report as new_reports.1, etc. E.g. \"{{num_counted_reports}} patients with {{alert_name}} in {{time_window_in_days}} days reported at {{new_reports.0.contact.parent.name}}. New reports from: {{new_reports.0.contact.name}}, {{new_reports.1.contact.name}}, {{new_reports.2.contact.name}}.\""
                         },
                         "recipients": {
                             "type": "array",
                             "title": "Recipients for the alerts",
-                            "description": "The phone numbers to receive alerts. You can use international format phone numbers (e.g. \"+254777888999\") or an expression to fetch phone fields in each counted_report (e.g. \"counted_report.contact.parent.contact.phone\" for the Primary Contact of the parent place of the sender.",
+                            "description": "The phone numbers to receive alerts. You can use international format phone numbers (e.g. \"+254777888999\") or an expression to fetch phone fields in each new_report (e.g. \"new_report.contact.parent.contact.phone\" for the Primary Contact of the parent place of the sender.",
```

Comment:
No, at this point only one report at a time is available. So you're going through the array of `new_reports`, and applying that `recipient` for each `new_report`. Thus the singular variable name: `new_report`, not `new_reports`. 

You don't have access to the whole array at that point, because of how the implementation is done, we're avoiding loading the whole array in memory because it could be big.

I understand where the confusion comes from though, the UX for this config is not the best, do you have thoughts on how to make it clearer?